spec/unit/functions/lookup_spec.rb

```diff
@@ -705,6 +708,25 @@ def explain(key, options = {})
           expect { lookup('mod::a') }.to raise_error(Puppet::DataBinding::LookupError, /all lookup_options patterns must match a key starting with module name/)
         end
       end
+
+      context 'and there are no lookup options that do not use patterns' do
```

Comment:
and "all lookup options use patterns" perhaps... to avoid the double negation. Not important enough to update the PR.